File: solutions/python/triangle/1/triangle.py
def equilateral(sides):
    a, b, c = sides
    return is_triangle(sides) and a == b == c

def isosceles(sides):
    a, b, c = sides
    return is_triangle(sides) and (a == b or b == c or a == c)

def scalene(sides):
    a, b, c = sides
    return is_triangle(sides) and (a != b and b != c and a != c)

def is_triangle(sides):
    a, b, c = sides
    return all([
        a > 0 and b > 0 and c > 0,
        a + b > c,
        b + c > a,
        a + c > b
    ])

print(equilateral([2, 2, 2]))   # True
print(isosceles([3, 3, 5]))     # True
print(scalene([3, 4, 5]))       # True
print(is_triangle([1, 2, 3]))   # False  

File: solutions/php/triangle/1/Triangle.php
<?php

declare(strict_types=1);

class Triangle
{
    private float $a;
    private float $b;
    private float $c;

    public function __construct(float $a, float $b, float $c)
    {
        // Validate that all sides are greater than 0
        if ($a <= 0 || $b <= 0 || $c <= 0) {
            throw new InvalidArgumentException('All sides must be greater than 0');
        }
        
        // Check triangle inequality: sum of any two sides must be >= third side
        if ($a + $b < $c || $a + $c < $b || $b + $c < $a) {
            throw new InvalidArgumentException('Violates triangle inequality');
        }
        
        $this->a = $a;
        $this->b = $b;
        $this->c = $c;
    }

    public function kind(): string
    {
        // Count how many unique side lengths we have
        $sides = [$this->a, $this->b, $this->c];
        $uniqueSides = array_unique($sides);
        
        if (count($uniqueSides) === 1) {
            return 'equilateral';
        }
        
        if (count($uniqueSides) === 2) {
            return 'isosceles';
        }
        
        return 'scalene';
    }
}